src/plugins/release-monitor/index.js

```diff
@@ -0,0 +1,292 @@
+                return pr;
+            }
+
+            return null;
+        })
+    )
+        .then(results => results.filter(result => result !== null));
+}
+
+/**
+ * Release label is present
+ * @param {Array<Object>} labels - collection of label objects
+ * @returns {boolean} True if release label is present
+ * @private
+ */
+function hasReleaseLabel(labels) {
+    return labels.some(({ name }) => name === RELEASE_LABEL);
+}
+
+/**
+ * Check if it is Post Release label
+ * @param {Object} label - label object
+ * @returns {boolean} True if its post release label
+ * @private
+ */
+function isPostReleaseLabel({ name }) {
+    return name === POST_RELEASE_LABEL;
+}
+
+/**
+ * Handler for issue label event
+ * @param {Object} context - probot context object
+ * @returns {Promise} promise
+ * @private
+ */
+async function issueLabeledHandler(context) {
+
+    // check if the label is post-release and the same issue has release label
+    if (isPostReleaseLabel(context.payload.label) && hasReleaseLabel(context.payload.issue.labels)) {
+
+        // put pending status on every PR which doesn't have fix or docs status.
+        await createStatusOnPRs({
+            context,
+            prs: await getNonSemverPatchPRs(context),
+            isSuccess: false
+        });
+    }
+}
+
+/**
+ * Handler for issue close event
+ * @param {Object} context - probot context object
+ * @returns {Promise} promise
+ * @private
+ */
+async function issueCloseHandler(context) {
+
+    // check if the closed issue is a release issue
+    if (hasReleaseLabel(context.payload.issue.labels)) {
+
+        // remove all the error status from any pr out their
+        await createStatusOnAllPRs({
+            context,
+            isSuccess: true
+        });
+    }
+}
+
+/**
+ * Handler for pull request open and reopen event
+ * @param {Object} context - probot context object
+ * @returns {Promise} promise
+ * @private
+ */
+async function prOpenHandler(context) {
+
+    /**
+     * check if the release issue has the label for no semver minor merge please
+     * false: add success status to pr
+     * true: add failure message if its not a fix or doc pr else success
+     */
+    const { data: releaseIssue } = await context.github.issues.getForRepo(
+        context.repo({
+            labels: `${RELEASE_LABEL},${POST_RELEASE_LABEL}`
+        })
+    );
+
+    const allCommits = await context.github.pullRequests.getCommits(context.issue());
+    const isSemverPatchPr = isMessageValidForPatchRelease(getCommitMessageForPR(allCommits, context.payload.pull_request));
+    const statusFunc =
+        releaseIssue.length === 0 || isSemverPatchPr
+            ? createSuccessPRStatus
+            : createPendingPRStatus;
+
+    await statusFunc({
+        context,
+        sha: pluckLatestCommitSha(allCommits),
+        description: isSemverPatchPr ? "This change is semver-patch" : undefined // eslint-disable-line no-undefined
```

Comment:
Less than nit: Would `void 0` be better and/or allow you to remove the eslint-disable-line comment?